upgrade.md

```diff
@@ -1,35 +1,126 @@
+
+## Setup permissions
+
+In case you are using Linux we highly recommend setting up the official project
+[udev rules](https://github.com/blackmagic-debug/blackmagic/tree/main/driver).
+
+```{warning}
+On Windows `bmputil-cli` will automatically take care of the needed drivers.
+Do **not** run Zadig on the DFU endpoint if using bmputil.
+```
+
+You can check if everything is working correctly by running:
+
+```sh
+bmputil-cli probe info
+```
+
+The tool should be able to find and list the Black Magic Probe connected to the system. Which will look something like
+this:
+
+```text
+Found: Black Magic Probe 2.0.0
+  Serial: 81C5797B
+  Port:  0-85000192
+```
+
+## Automatic Update
+
+This is the recommended procedure.
+
+```{note}
+This procedure is currently only supported by the [native](hardware.md#native-hardware) hardware. Third party
+hardware running Black Magic Firmware is not currently supported and the appropriate firmware has to be manually built.
+Refer to the [Manual Update](#manual-update) for instructions.
+```
+
+### Run update
+
+To upgrade the firmware on the connected Black Magic Probe all that should be necessary is running:
+
+```sh
+bmputil-cli probe update
+```
+
+And follow the instructions.
+
+If you would like to update the firmware to the latest release candidate you can run the following:
+
+```sh
+bmputil-cli probe update --use-rc
+```
+
+## Manual Update
+
+This procedure is necessary when the host platform is not the native hardware. We currently do not offer an automatic
+update path for third party hardware.
+
+This is also the procedure to follow if you are have some other reason to build the firmware manually. For example you
+are addinng new hardware support.
+
+### Download or build the firmware
 
 Download or compile the Black Magic Debug (BMD) firmware. Regarding firmware selection:
 
 * You can find the newest pre-built binaries on the
   [GitHub Release Page](https://github.com/blackmagic-debug/blackmagic/releases).
-* You can find the bleeding cutting edge binaries uploaded as assets on the
-  ["build and upload" GitHub actions page](https://github.com/blackmagic-debug/blackmagic/actions/workflows/build-and-upload.yml),
-  Click on the newest successful build and download the `blackmagic-firmware.zip` file. It contains binaries
-  for all the supported platforms.
-* When using the daily builds expect breaking changes. Please report issues on
+* You can download cutting edge binaries built with every commit to the main branch
+  [here](https://nightly.link/blackmagic-debug/blackmagic/workflows/build-and-upload/main).  When using the daily builds
+* expect breaking changes. Please report issues on
   [our issue page](https://github.com/blackmagic-debug/blackmagic/issues) or ask on our Discord server.
+* You can also build your own firmware from the cloned sources. Follow the [build instructions in the project
+* README](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building).
```

Comment:
Missing opening `[` here to make this a README link.

Reply:
how about now? :D

Reply:
That'll do the trick ^_^